r/R/query-engine.R

```diff
@@ -384,3 +392,61 @@ needs_projection <- function(projection, schema) {
     !identical(field_names, names(projection)) || # Any fields are renamed
     !identical(field_names, names(schema)) # The fields are reordered
 }
+
+#' Create projection needed to coalesce join keys after a full outer join
+#'
+#' @example
+#' test_join <- list(
+#'   type = JoinType$FULL_OUTER,
+#'   right_data = arrow_table(x = 1, y = 2, z = "x"),
+#'   by = c("x", 'y'),
+#'   suffix = c(".x", ".y"),
+#'   keep = FALSE
+#' )
+#' post_join_projection(c("value", "x", "y", "z"), test_join)
```

Comment:
I may be misremembering, but would you mind checking if including this header generates an .Rd file?  If so (and assuming the point is here that this is an internal function), we should use the @noRd tag (or whatever equivalent we use in other non-exported functions with Roxygen headers) to ensure we're not generating user documentation.

Reply:
It does, but I can add `@noRd`.